### PR TITLE
chore: Migrate gatsby-starter-mdx-basic to monorepo

### LINKED ISSUE #16350: Move gatsby-starter-mdx-basic into gatsby repo (or other solution)
Via: https://github.com/gatsbyjs/gatsby/pull/16348 The repo (https://github.com/ChristopherBiscardi/gatsby-starter-mdx-basic) is still under my namespace which doesn't really make sense any more now that gatsby-plugin-mdx is in the gatsby repo. We can either port that starter over or create a new one that we feel is better suited for "getting started with MDX quickly".

cc/ @johno @gatsbyjs/learning

### PR DESCRIPTION
**Pull request by laurieontech**
Fixes #16350

### CODE REVIEW

File: starters/gatsby-starter-mdx-basic/LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
```
Comment (LekoArts) [inline starters/gatsby-starter-mdx-basic/LICENSE:1]: We recently changed the licenses of our starters to be 0BSD (see https://github.com/gatsbyjs/gatsby/pull/25441). That change will be necessary here

Comment (laurieontech) [reply starters/gatsby-starter-mdx-basic/LICENSE:1]: Thanks!

File: starters/gatsby-starter-mdx-basic/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Gatsby Starter MDX Basic
+
+View this starter [live on Netlify](https://gatsby-starter-mdx-basic.netlify.com/)
```
Comment (LekoArts) [inline starters/gatsby-starter-mdx-basic/README.md:3]: Is this still the URL from chris or ours?

File: starters/gatsby-starter-mdx-basic/gatsby-config.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      options: {
+        name: `images`,
+        path: `${__dirname}/src/images`,
+      },
+    },
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `pages`,
+        path: `${__dirname}/src/pages`
```
Comment (LekoArts) [inline starters/gatsby-starter-mdx-basic/gatsby-config.js:29]: iirc we source src/pages for MDX files automatically now?

File: starters/gatsby-starter-mdx-basic/src/components/image.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * `StaticQuery`, which allows us to load the image from directly within this
+ * component, rather than having to pass the image data down from pages.
+ *
+ * For more information, see the docs:
+ * - `gatsby-image`: https://gatsby.app/gatsby-image
+ * - `StaticQuery`: https://gatsby.app/staticquery
+ */
+
+const Image = () => (
+  <StaticQuery
```
Comment (LekoArts) [inline starters/gatsby-starter-mdx-basic/src/components/image.js:17]: Should be a useStaticQuery IMO

File: starters/gatsby-starter-mdx-basic/README.md (hunk 2 above)
Comment (laurieontech) [reply starters/gatsby-starter-mdx-basic/README.md:3]: Chris. I was going to keep it and then I can redeploy? Or I could just remove it.

Comment (LekoArts) [reply starters/gatsby-starter-mdx-basic/README.md:3]: That's probably on his account so you can't change the source? 🤔 Then just remove it for now

File: starters/gatsby-starter-mdx-basic/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Gatsby Starter MDX Basic
+
+This starter build MDX support into the
+[gatsby-default-starter](https://github.com/gatsbyjs/gatsby-starter-default). Its
+README also applies here.
+
+1. Create a new Gatsby site using this starter
+
+```sh
+gatsby new my-site https://github.com/ChristopherBiscardi/gatsby-starter-mdx-basic
```
Comment (LekoArts) [inline starters/gatsby-starter-mdx-basic/README.md:10]:
```suggestion
gatsby new my-site https://github.com/gatsbyjs/gatsby-starter-mdx-basic
```

**Pull request merged**